scripts/clean_appliance.py

```diff
@@ -38,14 +42,30 @@ def main():
     }
     if args.hostname is not None:
         ssh_kwargs['hostname'] = args.hostname
+        appliance = IPAppliance(args.hostname)
+        if appliance.version < '5.9':
+            # Running evm:db:reset on CFME 5.8 sometimes leaves it in a state
+            # where it is unable to start again
+            print('EXITING: This script does not work reliably for CFME 5.8')
+            return 1
 
-    with SSHClient(stream_output=True, **ssh_kwargs) as ssh_client:
+    with SSHClient(stream_output=not args.quiet, **ssh_kwargs) as ssh_client:
         # Graceful stop is done here even though it is slower than killing ruby processes.
         # Problem with killing ruby is that evm_watchdog gets spawned right after being killed
         # and then prevents you from destroying the DB.
         # Graceful stop makes sure there are no active connections to the DB left.
         print('Stopping evmserverd...')
         ssh_client.run_command('systemctl stop evmserverd')
+
+        @wait_for_decorator(num_sec=60, delay=5)
+        def check_no_db_connections():
```

Comment:
I don't see this subroutine being called anywhere? Is it being called anywhere?

Reply:
It is called thanks to the decorator. If you use that decorator, below function will get called wrapped in `wait_for`.

Reply:
Cool, thanks I understand now.